### ISSUE TITLE
Missing search files in QCH content

### ISSUE THREAD

**Issue by sjoubert** [doxygen/doxygen#9816]
**Describe the bug**
The QCH file generated seems to be missing some files related to the search functionality. In particular the `search/searchdata.js` and lots of `search/*.{html,js}` pair. I don't know which other files are missing, but those are the ones I noticed.

**Expected behavior**
The QCH database should package all referenced files.

**Screenshots**
Here is a list of all files in the "search/" folder present in the QCH database and the content of the same folder in the HTML output:
![Screenshot_20230126_103607](https://user-images.githubusercontent.com/200310/214803256-2b8df4b9-6cbf-4319-b3fd-126915df8ac3.png)
![Screenshot_20230126_104031](https://user-images.githubusercontent.com/200310/214804113-d0d5e2bc-c4b1-4875-8ed2-cdf0f8db2736.png)

**To Reproduce**
I generate HTML and QHP output at the same time using a single Doxyfile. Not sure which configuration option/value would be relevant here.

I use the QCH file to display the documentation content in a Qt application through a QWebEngineView (and a custom qthelp:// scheme handler to fetch the data to display from the QCH file). When displaying the home (or any other) page I get the following error log:
```
js: jQuery.Deferred exception: indexSectionLabels is not defined ReferenceError: indexSectionLabels is not defined
    at init_search (qthelp://com.example/doc/search/search.js:791:19)
    at HTMLDocument.<anonymous> (qthelp://com.example/doc/index.html:45:34)
    at e (qthelp://com.example/doc/jquery.js:2:30038)
    at t (qthelp://com.example/doc/jquery.js:2:30340) undefined
js: Uncaught ReferenceError: indexSectionLabels is not defined
```
This `indexSectionLabels` variable is defined in the `search/searchdata.js` file which is not present in the QCH file. In addition to the error log in the console this breaks the search box (which doesn't respond to input anymore).

**Version**
1.9.4 from my Debian installation, but also reproduced with a manual installation of 1.9.6

**Comment by albert-github** [comment]
I tried to reproduce the problem and it looks like that the search box is indeed not correct (it is hard to test as I don't have a recent webviewer as it isn't distributed anymore by Qt, to the best of mt knowledge, and creating one myself failed till now).

I created a small project (see the: [example.tar.gz](https://github.com/doxygen/doxygen/files/10508448/example.tar.gz))

and this generates a `index.qhp` and it looks like that (in this case, the following files are missing:
```
search/all_0.js
search/all_1.js
search/files_0.js
search/functions_0.js
search/searchdata.js
```

With the `index_extended.qhp` and running it, in my case through the `qhelpgenerator` and `assistent` of a Qt 5.4.1 version the functionality seems to work.

- what are your settings in the Doxyfile and attach the result as  a, compressed, tar or zip file.
- can you add the `*.js` files (and `*.html` and other missing files from the `search` directory) to the `qhp` file and see whether or not it works?

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9818 

Only the `*.js` files are added, in the 1.9.4 version there were also some `*.html` files created but these are not present / necessary anymore since version 1.9.5 (so won't be added either). 


Only the `*.js`

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
